pkg/provider/handlers/deploy_test.go

```diff
@@ -0,0 +1,63 @@
+package handlers
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/openfaas/faas-provider/types"
+)
+
+func Test_BuildLabels_WithAnnotations(t *testing.T) {
+	// Test each combination of nil/non-nil annotation + label
+	tables := []struct {
```

Comment:
This can be improved by also adding a name to each test case and using a subtest

```go
    // Test each combination of nil/non-nil annotation + label
	tables := []struct {
        name string
		label      map[string]string
		annotation map[string]string
		result     map[string]string
	}{
		{"empty annotations returns empty lable map", nil, nil, map[string]string{}},
		{"...",map[string]string{"L1": "V1"}, nil, map[string]string{"L1": "V1"}},
		{"....",nil, map[string]string{"A1": "V2"}, map[string]string{fmt.Sprintf("%sA1", annotationLabelPrefix): "V2"}},
		{
            "....",
			map[string]string{"L1": "V1"}, map[string]string{"A1": "V2"},
			map[string]string{"L1": "V1", fmt.Sprintf("%sA1", annotationLabelPrefix): "V2"},
		},
	}
    for _, pair := range tables {
        t.Run(pair.name, func(t *testing.T){
            request := &types.FunctionDeployment{
                Labels:      &pair.label,
                Annotations: &pair.annotation,
            }
            val, err := buildLabels(request)
            if err != nil {
                t.Fatalf("want: no error got: %v", err)
            }
            if !reflect.DeepEqual(val, pair.result) {
                t.Errorf("Got: %s, expected %s", val, pair.result)
            }
        })
	}
```

Reply:
Thanks for the feedback - I've added the test name and included more newlines to make the test table a bit more readable and clear.